app/controllers/evaluations_controller.rb

```diff
@@ -86,6 +86,21 @@ def cancel_point_request
                 notice: t(:cancelled_point_request)
   end
 
+  def copy_previous_principles
+    authorize_evaluation
+
+    previous_evaluation = current_group.evaluations.second_to_last
+    previous_evaluation.principles.each do |principle|
+      Principle.create!(evaluation: current_evaluation,
+                        name: principle.name,
+                        description: principle.description,
+                        type: principle.type,
+                        max_per_member: principle.max_per_member)
+    end
```

Comment:
Have you considered using a transaction here? Now, it can result in an unexpected state if the copy fails for some reason.

```ruby
ActiveRecord::Base.transaction do
  # ...
end
```

Also, I'd recommend using `#find_each` instead of `#each`. It's less important right now, but it's more future-proof.